### ISSUE TITLE
ovn-controller with ovs-dpdk ovn-controller.log has errors “Invalid or incomplete DHCP packet received”

### ISSUE THREAD

**Issue by AaronChu0** [ovn-org/ovn#82]
ovn-controller（ovn-branch-20.06） Keep reporting errors in the function "pinctrl_handle_put_dhcp_opts"(pinctrl.c) .The error message is “Invalid or incomplete DHCP packet received”。I use OVN on the basis of ovs-dpdk, so the struct dp_packet member of the structure should use mbuf, but ovn does not have the macro “DPDK_NETDEV”, which makes it impossible to parse the message normally.
I found a temporary solution, define “DPDK_NETDEV” when ovn uses ovs lib struct dp_packet, and then add the dpdk header file to ovn, so that packetin dhcp packets can be parsed normally.
![image](https://user-images.githubusercontent.com/80369022/110561514-c40f4a00-8182-11eb-8017-7a404bb2640f.png)

**Comment by blp** [comment]
Oh, this is nasty. It's config.h in OVS that defines DPDK_NETDEV, but OVN doesn't (and shouldn't) #include OVS's config.h.

**Comment by AaronChu0** [comment]
I currently plan to add with-dpdk to OVN configure. Do you have a better solution?

**Comment by blp** [comment]
Yes. OVN configure should determine whether OVS was configured to use DPDK. OVN configure has everything it needs to do that, since it already knows the OVS build directory.
